src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/basm/BasmService.kt

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.digital.hmpps.welcometoprison.model.basm
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.welcometoprison.model.MoveType
+import uk.gov.justice.digital.hmpps.welcometoprison.model.Movement
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.Includes
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.Location
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.People
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.Profile
+import java.time.LocalDate
+import kotlin.reflect.KClass
+
+@Service
+class BasmService(private val basmClient: BasmClient) {
+
+  fun <T : Includes> Map<String, Map<String, Includes>>.getAs(key: String, type: KClass<T>): (id: String) -> T? {
+    val includesOfType = this[key] ?: emptyMap()
+    return { id ->
+      val item = includesOfType[id]
+      check(item == null || type.isInstance(item)) { "item expected to be $type, was ${item?.javaClass}" }
+      @Suppress("UNCHECKED_CAST")
+      item as T?
+    }
+  }
+
+  fun getMoves(prisonId: String, fromDate: LocalDate, toDate: LocalDate): List<Movement> {
+
+    val prison = basmClient.getPrison(prisonId) ?: return emptyList()
```

Comment:
have to look up prison by agency id to then use it's basm UUID to filter moves by location